### PR TITLE
Framed glasslike: Don't use cuboids to draw glass faces

### PR DESCRIPTION
**Pull request by paramat**
Commit 1:
Framed glasslike: Don't use cuboids to draw glass faces

Previously, each glass face used drawAutoLightedCuboid() to draw a
flat cuboid. This also disallowed backface culling, making the
backface culling inconsistent with 'glasslike'.

Use code from 'glasslike' to draw glass faces using drawQuad().
//////////////////////
Commit 2:
Framed glasslike: Remove long-unknown top/bottom textures feature

Makes the code simpler and cleaner.
Never documented, long-unknown and not of much use. We are already
considering making this drawtype accept 6 textures for the 6 faces,
which will be far more useful.
///////////////////////

screenshot_20181102_220235

^ Smooth lighting disabled

screenshot_20181102_220140

^ Smooth lighting enabled

screenshot_20181102_230640

^ Backface culling can be disabled for the glass faces, no z-fighting

Fixes remaining bug of #7823 : lack of backface culling.
See https://github.com/minetest/minetest/issues/7823#issuecomment-434794055 onwards for more details.

For reviewers:

Instead of using `drawAutoLightedCuboid()` to draw each glass face, code from `drawGlasslikeNode()` (which is directly above) is copy-pasted.
The only changes are:
* Using the slightly inset distance `g` for vertexes to avoid z-fighting against adjacent nodes if backface culling is disabled.
* Because `drawGlasslikeFramedNode()` uses `getTile()` instead of `useTile()` it has this code from `useTile()` added for correct lighting:
```
	if (!data->m_smooth_lighting)
		color = encode_light(light, f->light_source);
```

See #1428 . The addition of top/bottom textures was not mentioned or discussed in the PR. "add interior volume." refers to the internal liquid feature, and both new features were not documented, sloppy, it's good to know our quality and required standards are higher now.

### CODE REVIEW

File: src/content_mapblock.cpp [hunk 1]
```diff
@@ -800,8 +802,34 @@ void MapblockMeshGenerator::drawGlasslikeFramedNode()
...
+		for (v3f &vertex : vertices) {
+			switch (face) {
+				case D6D_ZP:
+					vertex.rotateXZBy(180); break;
+				case D6D_YP:
+					vertex.rotateYZBy( 90); break;
+				case D6D_XP:
+					vertex.rotateXZBy( 90); break;
+				case D6D_ZN:
+					vertex.rotateXZBy(  0); break;
```
Comment (SmallJoker) [inline src/content_mapblock.cpp]: Unnecessary transformation.
```suggestion
					break;
```

**Comment by numberZero** [comment]
@paramat For flat lighting, you need to set color, like in `useTile`:
`color = encode_light(light, f->light_source);`

File: src/content_mapblock.cpp (hunk 1 above)
Comment (paramat) [reply src/content_mapblock.cpp]: This was copied from 'glasslike', but yes will consider.

**Comment by paramat** [comment]
I see, thanks, working on it.

**Comment by paramat** [comment]
Getting z-fighting between frame and any full-width glass face texture used, such as top/base textures:

screenshot_20181102_224908

Will fix.

**Comment by paramat** [comment]
Meh that will be messy, i think i prefer to remove the top/bottom feature, it was undocumented so we can do that, it's not very useful, and i can't remember anyone being aware of it. We are already considering making framed glasslike accept 6 textures, which would be much more useful.

**Comment by paramat** [comment]
@SmallJoker i've added some info to the 1st post to help with review, this is semi-trivial, i'm wondering if you can approve?

**Pull request merged**